datastore/postgres/migrations/migrations_test.go

```diff
@@ -0,0 +1,67 @@
+package migrations
+
+import (
+	"bufio"
+	iofs "io/fs"
+	"os"
+	"path"
+	"regexp"
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestMigrationsMismatch(t *testing.T) {
+	var migrations, files []string
+
+	// Get referenced migrations
+	migrationLine, err := regexp.Compile(`runFile\(\"(.*)\"\)`)
+	if err != nil {
+		t.Fatal(err)
+	}
+	f, err := os.Open("migrations.go")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		ms := migrationLine.FindSubmatch(s.Bytes())
+		switch {
+		case ms == nil, len(ms) == 1:
+			continue
+		case len(ms) == 2:
+			migrations = append(migrations, path.Clean(string(ms[1])))
+		}
+	}
+	if err := s.Err(); err != nil {
+		t.Error(err)
+	}
+	sort.Strings(migrations)
+
+	// Get migration files
+	err = iofs.WalkDir(os.DirFS("."), ".", func(p string, d iofs.DirEntry, err error) error {
+		switch {
+		case err != nil:
+			return err
+		case d.IsDir():
+			return nil
+		case path.Ext(d.Name()) != ".sql":
+			return nil
+		}
+		files = append(files, p)
+		return nil
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	sort.Strings(files)
+
+	// Check referenced files exist and existing files are referenced.
+	if !cmp.Equal(migrations, files) {
```

Comment:
I believe this will fail on Windows machines due to the `\`? If so, is that ok?

Reply:
We don't support running on Windows so I think it's ok